In [28]:
from math import factorial, e

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

def permutations(n):
    return int(factorial(n))

### Задание 1
Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Воспользуемся формулой Бернулли.
$$ p(k) = C_n^k p^k q^{n-k} $$
k = 85, p = 0.8, q = 1 - 0.8, n = 100

In [29]:
p_k85 = combinations(100, 85) * pow(0.8, 85) * pow(0.2, 15)
p_k85

0.048061793700746556

Ответ: 0.048061793700746556

### Задание 2
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Воспользуемся распределением Пуассона.
$$ P_m \approx \frac{\lambda^m }{m!}e^{-\lambda}$$
n = 5000, p = 0.0004, m = 0 для первого случая и m = 2 для второго случая
$$ \lambda = np $$

In [30]:
l = 5000 * 0.0004
P_0 = pow(l, 0) / factorial(0) * pow(e, -l)
P_0

0.1353352832366127

In [31]:
P_2 = pow(l, 2) / factorial(2) * pow(e, -l)
P_2

0.2706705664732254

Ответ:

а) 0.1353352832366127

б) 0.2706705664732254

### Задание 3
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Воспользуемся формулой Бернулли
k = 70, p = 0.5, q = 0.5, n = 144

In [32]:
p_k70 = combinations(144, 70) * pow(0.5 , 70) * pow(0.5, 74)
p_k70

0.06281178035144776

Ответ: 0.06281178035144776

### Задание 4
В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

а) Все мячи белые
Вероятность вытащить белый мяч из первого ящика 7/10, второй мяч 6/9. Вероятность вытащить белый мяч из второго ящика 9/11, второй мяч 8/10

In [33]:
P = 7/10 * 6/9 * 9/11 * 8/10
P

0.3054545454545454

б) Вероятность, того что 2 мяча белых.
Вариант 1 - все белые мячи из первого ящика
Вариант 2 - все белые мячи из второго ящика
Вариант 3 - 1 белый мяч из первого ящика и 1 белый мяч из второго ящика

In [34]:
P_1 = 7/10 * 6/9 * 2/11 * 1/10
P_1

0.008484848484848482

In [35]:
P_2 = 9/11 * 8/10 * 3/10 * 2/9
P_2

0.04363636363636363

In [36]:
P_3 = 7/10 * 3/9 * 9/11 * 2/10
P_3

0.03818181818181818

In [37]:
P = P_1 + P_2 + P_3
P

0.0903030303030303

в) Вероятность того, что хотя бы один мяч белый.
Тут найдем вероятность, когда нет вообще белого мяча и вычтем из 1 эту вероятность

In [38]:
P_black = 3/10 * 2/9 * 2/11 * 1/10
P_black

0.0012121212121212121

In [39]:
P = 1 - P_black
P

0.9987878787878788

Ответ:

а) 0.3054545454545454

б) 0.0903030303030303

в) 0.9987878787878788